controllers/spacerequest/spacerequest_controller.go

```diff
@@ -43,12 +63,440 @@ func (r *Reconciler) SetupWithManager(mgr ctrl.Manager, memberClusters map[strin
+		Namespace: r.Namespace,
+		Name:      tierName,
+	}, tier); err != nil {
+		if errors.IsNotFound(err) {
+			return err
+		}
+		// Error reading the object - requeue the request.
+		return errs.Wrap(err, "unable to get the current NSTemplateTier")
+	}
+	return nil
+}
+
+// updateSubSpace updates the tierName and targetClusterRoles from the spaceRequest to the subSpace object
+// if they are not up to date.
+// returns false/nil if everything is up to date
+// returns true/nil if subSpace was updated
+// returns false/err if something went wrong
+func (r *Reconciler) updateSubSpace(logger logr.Logger, subSpace *toolchainv1alpha1.Space, spaceRequest *toolchainv1alpha1.SpaceRequest) (bool, error) {
+	logger.Info("update subSpace")
+	if spaceRequest.Spec.TierName == subSpace.Spec.TierName &&
+		reflect.DeepEqual(spaceRequest.Spec.TargetClusterRoles, subSpace.Spec.TargetClusterRoles) {
+		// everything is up to date let's return
+		return false, nil
+	}
+
+	// update tier name and target cluster roles
+	logger.Info("updating subSpace", "name", subSpace.Name)
+	subSpace.Spec.TierName = spaceRequest.Spec.TierName
+	subSpace.Spec.TargetClusterRoles = spaceRequest.Spec.TargetClusterRoles
+	err := r.Client.Update(context.TODO(), subSpace)
+	if err != nil {
+		return false, errs.Wrap(err, "unable to update tiername and targetclusterroles")
+	}
+
+	logger.Info("subSpace updated", "name", subSpace.Name, "target_cluster_roles", subSpace.Spec.TargetClusterRoles, "tierName", subSpace.Spec.TierName)
+	return true, nil
+}
+
+// getParentSpace retrieves the name of the space that provisioned the namespace in which the spacerequest was issued.
+// we need to find the parentSpace so we can `link` the 2 spaces and inherit the spacebindings from the parent.
+func (r *Reconciler) getParentSpace(memberCluster cluster.Cluster, spaceRequest *toolchainv1alpha1.SpaceRequest) (*toolchainv1alpha1.Space, error) {
+	parentSpace := &toolchainv1alpha1.Space{}
+	namespace := &v1.Namespace{}
+	err := memberCluster.Client.Get(context.TODO(), types.NamespacedName{
+		Namespace: "",
+		Name:      spaceRequest.Namespace,
+	}, namespace)
+	if err != nil {
+		return parentSpace, errs.Wrap(err, "unable to get namespace")
+	}
+	// get owner name which is equal to NSTemplateSet name and Space name
+	parentSpaceName, found := namespace.Labels[toolchainv1alpha1.OwnerLabelKey]
+	if !found || parentSpaceName == "" {
+		err := errs.Errorf("unable to find owner label %s on namespace %s", toolchainv1alpha1.OwnerLabelKey, namespace.GetName())
+		return parentSpace, err
+	}
+
+	// check that parentSpace still exists and is not being deleted
+	err = r.Client.Get(context.TODO(), types.NamespacedName{
+		Namespace: r.Namespace,
+		Name:      parentSpaceName,
+	}, parentSpace)
+	if err != nil {
+		if errors.IsNotFound(err) {
+			// parentSpace was deleted
+			return parentSpace, errs.Wrap(err, "the parentSpace does not exist")
+		}
+		return parentSpace, errs.Wrap(err, "unable to get parentSpace")
```

Comment:
let's just simplify it
```suggestion
		return parentSpace, errs.Wrap(err, "unable to get parentSpace")
```